packages/trace-viewer/src/sw.ts

```diff
@@ -145,6 +146,49 @@ async function doFetch(event: FetchEvent): Promise<Response> {
   return snapshotServer.serveResource(lookupUrls, request.method, snapshotUrl);
 }
 
+function headersForResource(traceModel: TraceModel, sha1: string): Headers | undefined {
+  const attachment = traceModel.attachmentForSha1(sha1);
+  if (!attachment)
+    return;
+  const headers = new Headers();
+  let name = attachment.name;
+  const nameHasExt = name.includes('.');
+  if (!nameHasExt)
+    name += extensionForContentType(attachment.contentType);
```

Comment:
Was this explicitly requested? I'd leave `attachment.name` without adding anything.

Reply:
@mxschmitt add a comment explaining why we add it here?